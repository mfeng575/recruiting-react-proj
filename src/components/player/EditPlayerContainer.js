import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import toastr from 'toastr';
import * as playerAction from '../../action/PlayerAction';
import PlayerForm from './PlayerForm'; 

export class EditPlayerContainer extends React.Component {

    constructor() {
        super();
        this.handleSave = this.handleSave.bind(this);
        this.handleCancel = this.handleCancel.bind(this);
    }

    componentDidMount() {
        this.props.action.getPlayerAction && this.props.action.getPlayerAction(this.props.match.params.id)
            .catch(error => {
                toastr.error(error);
            });
    }

    handleSave(values) {
        const player = {
            id: values.id,
            firstName: values.firstName,
            lastName: values.lastName,
            score: values.score
        };

        this.props.action.savePlayerAction(player)
            .then(() => {
                toastr.success('Player saved');
                this.props.history.push('/leaderboard');
            }).catch(error => {
                toastr.error(error);
            });
    }

    handleCancel(event) {
        event.preventDefault();
        this.props.history.replace('/leaderboard');
    }

    render() {
        const { initialValues } = this.props;
        const heading = initialValues && initialValues.id ? 'Edit Player' : 'Add Player';

        return (
            <div className="container">
                <PlayerForm
                    heading={heading}
                    handleSave={this.handleSave}
                    handleCancel={this.handleCancel}
                    initialValues={this.props.initialValues}
                />
            </div>
        );
    }
}

const mapStateToProps = (state, ownProps) => {
    const playerId = ownProps.match.params.id; //from the path '/player/:id'

    if (playerId && state.selectedPlayerReducer.player && playerId === state.selectedPlayerReducer.player.id) {
        return {
            initialValues: state.selectedPlayerReducer.player
        };
    } else {
        return {
        };
    }
};

const mapDispatchToProps = dispatch => ({
    action: bindActionCreators({ ...playerAction }, dispatch)
});


EditPlayerContainer.propTypes = {
    action: PropTypes.object.isRequired,
    history: PropTypes.object,
    initialValues: PropTypes.object,
    match: PropTypes.object.isRequired
};



export default connect(mapStateToProps, mapDispatchToProps)(EditPlayerContainer);
