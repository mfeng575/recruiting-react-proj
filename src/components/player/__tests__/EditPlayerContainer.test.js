import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import toJson from 'enzyme-to-json';
import {EditPlayerContainer} from '../EditPlayerContainer';


describe('EditPlayerContainer.test.js', () => {

    it('renders without crashing', () => {
        const props = {
            action: {
            },
            initialValues: {id: '', firstName: '', lastName: '', score: 0},
            match: {params: {id:'1'}}
        };

        const wrapper = shallow(<EditPlayerContainer {...props}/>);
        expect(wrapper.length).toEqual(1);

        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

});



    
        