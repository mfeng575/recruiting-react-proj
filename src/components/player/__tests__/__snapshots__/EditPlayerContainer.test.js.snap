// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPlayerContainer.test.js renders without crashing 1`] = `
<div
  className="container"
>
  <ReduxForm
    destroyOnUnmount={true}
    enableReinitialize={false}
    forceUnregisterOnUnmount={false}
    form="PlayerForm"
    getFormState={[Function]}
    handleCancel={[Function]}
    handleSave={[Function]}
    heading="Add Player"
    initialValues={
      Object {
        "firstName": "",
        "id": "",
        "lastName": "",
        "score": 0,
      }
    }
    keepDirtyOnReinitialize={false}
    persistentSubmitErrors={false}
    pure={true}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    touchOnBlur={true}
    touchOnChange={false}
    updateUnregisteredFields={false}
  />
</div>
`;
